src/Provider/DefaultSmsProvider.php

```diff
@@ -216,6 +203,8 @@ public function processDeliveryReport(Request $request, SmsGatewayInterface $sms
    * @return \Drupal\sms\Entity\SmsGatewayInterface|null
    *   A SmsGateway config entity, or NULL if default gateway is not set or
    *   invalid.
+   *
+   * @deprecated remove. shouldnt rely on this since its really a fallback.
```

Comment:
Im not saying the default (fallback) gateway should go, but the method shouldnt be relied on by APIs. it shouldnt be first class. I have made it an implementation detail of ensureGateways